### PR TITLE
Improved save_result

### PR DESCRIPTION
**Pull request by clausmichele**
I did the following changes:
1. Aligned output dimensions with 'x','y','t' for netCDF
2. Check lowercase format for netCDF, allowing 'netCDF', 'NETCDF' etc.
3. netCDFs now have band names corresponding to datacube bands and not 'stack-....'
4. Allow output format to be any of 'gtiff','geotiff' checking the lowercase of the string for geoTIFF
5. geoTIFFs now have band names corresponding to datacube bands and not 'stack-....'

### CODE REVIEW

File: src/openeo_processes/cubes.py [hunk 1]
```diff
@@ -263,12 +290,26 @@ def exec_xar(data, output_filepath='out', format='GTiff', options={}):
             if hasattr(data, 'time') and hasattr(data.time, 'units'):
                 data.time.attrs.pop('units', None)
             # end workaround
+            
+            data = refactor_data(data)
             data.to_netcdf(path=output_filepath)
-        elif format == 'GTiff':
+
+        elif format.lower() in ['gtiff','gtif','geotiff','geotif','tiff','tif']:
```
Comment (m-mohr) [inline src/openeo_processes/cubes.py:297]:
```suggestion
        elif format.lower() in ['gtiff','geotiff']:
```

GTiff is the only allowed option if following the API spec closely. I could see that people type geotiff by accident, but I wouldn't allow too many options here for interoperability reasons.

Comment (clausmichele) [reply src/openeo_processes/cubes.py]: Ok fine, I'll keep only the two options as you suggest

**Review by sophieherrmann — COMMENTED** [review]
Generally looks good, but xarray is not imported, so xr is not defined. Please add the import.

**Comment by clausmichele** [comment]
> 
> 
> Generally looks good, but xarray is not imported, so xr is not defined. Please add the import.

Done!

**Comment by sophieherrmann** [comment]
I just saw `rioxarray` is missing under `tests_require` and two tests fail for me: `test_save_result` and `test_apply_kernel` - I think for the later again a xarray import is missing.

**Pull request merged**